content/en/blog/_posts/TBD-admission-controllers-for-container-drift/index.md

```diff
@@ -0,0 +1,90 @@
+
+Here comes a sequence diagram of the entire workflow mentioned above: 
+{{< figure src="https://github.com/box/kube-exec-controller/raw/main/workflow-diagram.png" alt="Workflow Diagram" width="75%" >}}
+
+## A new kubectl plugin for better user experience
+Our admission controller service works great for solving the container drift issue we had on the platform. It is able able to submit all related K8s events to the target pod that has been affected. However, as K8s events often get away quickly (one hour of persistency by the default configuration), we need to provide other ways for developers to get their pod interaction activity. A [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) is a perfect choice for us to expose this information. We name our plugin “kubectl-pi” (short for pod-interaction) and provide two subcommands: “get” and “extend”. When the “get” command is called, the plugin simply checks the metadata attached by our admission controller service and transfers it to human-readable information. Here is an example output of the `kubectl pi get` command:
+
+```
+$ kubectl pi get test-pod
+POD_NAME  INTERACTOR  POD_TTL  EXTENSION  EXTENSION_REQUESTER  EVICTION_TIME
+test-pod  username-1  1h0m0s   /          /                    2021-09-24 17:00:00 -0800 PST
+```
+
+The plugin can also be used to extend the current eviction time of a pod, in case developers need more buffer to debug ongoing issues after executing on their containers. This is achieved by the `kubectl pi extend` command, where the plugin patches relevant K8s annotation to the given pod. This annotation includes the duration and username who made the extension request for transparency (also displayed in the table returned from the “get” command). There is another webhook defined in our admission controller service that admits annotation updates and resets the eviction timer of the target pod accordingly. An example of requesting the extension from the developer side would be:
```

Comment:
```suggestion
The plugin can also be used to extend the TTL for a pod that is marked for future eviction. This is useful in case developers need extra time to debug ongoing issues. To achieve this, a developer uses the `kubectl pi extend` subcommand, where the plugin patches the relevant annotation for the given pod. This annotation includes the duration and username who made the extension request for transparency (also displayed in the table returned from the “get” command).

There is another webhook defined in kube-exec-controller. This webhook admits valid annotation updates; once aditted, those updates reset the eviction timer of the target pod accordingly. An example of requesting the extension from the developer side would be:
```

Reply:
Updated based on the suggestion above (with slight rephrasing).